core/src/main/kotlin/fr/sncf/osrd/signal_projection/SignalProjection.kt

```diff
@@ -277,6 +277,11 @@ private fun signalUpdates(
         for (i in 0 until events.size - 1) {
             val event = events[i]
             val nextEvent = events[i + 1]
```

Comment:
As there is no special filter for the last event anymore, the code can probably be mutualized.
Something along this (there is probably something more idiomatic to be done):
```suggestion
            val endTime = if (i < events.size) events[i + 1].time else simulationEndTime
```